### PR TITLE
make sure `repo_dir` is empty before 'dump-repo'

### LINKED ISSUE #20191: `dump-repo` command is somewhat dangerous: removing the data directory if failed.
DumpRepository -> migrateRepository -> uploader.Rollback -> os.RemoveAll(g.baseDir)

If migrateRepository returns error, then the data directory is removed completely.

I have run `gitea dump-repo` in the gitea directory, then the Gitea's data directory (including sqlite database, repositories) is removed, all lost (luckily not on production).

### PR DESCRIPTION
**Pull request by a1012112796**
or it looks dangerous ...

fix #20191

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

File: cmd/dump_repo.go [hunk 1]
```diff
@@ -159,9 +160,14 @@ func runDumpRepository(ctx *cli.Context) error {
 		}
 	}
 
+	repoDir := ctx.String("repo_dir")
+	if dir, _ := os.ReadDir(repoDir); len(dir) > 0 {
```
Comment (lunny) [inline cmd/dump_repo.go]: repoDir could be not exist or empty directory.

Comment (wxiaoguang) [reply cmd/dump_repo.go]: If repoDir doesn't exist, the `len(dir)` is 0? Add more comments?

Comment (wxiaoguang) [reply cmd/dump_repo.go]: Let's add a comment and move on (actually, added more checks)

**Review by wxiaoguang — APPROVED** [review]

**Comment by 6543** [comment]
-> #20370

**Pull request merged**